src/helpers/abi/utils.ts

```diff
@@ -0,0 +1,215 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+
+const EXPLORER_API_URLS = {
+  '1': 'https://api.etherscan.io/api',
+  '4': 'https://api-rinkeby.etherscan.io/api'
+};
+
+const GNOSIS_SAFE_TRANSACTION_API_URLS = {
+  '1': 'https://safe-transaction.gnosis.io/api/v1/',
+  '4': 'https://safe-transaction.rinkeby.gnosis.io/api/v1/'
+};
+
+const ERC20ContractABI = [
+  'function transfer(address recipient, uint256 amount) public virtual override returns (bool)'
+];
+
+const ERC721ContractABI = [
+  'function safeTransferFrom(address _from, address _to, uint256 _tokenId) external payable'
+];
+
+const MULTISEND_CONTRACT_ADDRESS = '0x8D29bE29923b68abfDD21e541b9374737B49cdAD';
+
+const fetchContractABI = memoize(
+  async (url: string, contractAddress: string) => {
+    const params = new URLSearchParams({
+      module: 'contract',
+      action: 'getAbi',
+      address: contractAddress
+    });
+
+    const response = await fetch(`${url}?${params}`);
+
+    if (!response.ok) {
+      return { status: 0, result: '' };
+    }
+
+    return response.json();
+  },
+  (url, contractAddress) => `${url}_${contractAddress}`
+);
+
+export const mustBeEthereumAddress = memoize((address: string) => {
+  const startsWith0x = address?.startsWith('0x');
+  const isValidAddress = isAddress(address);
+  return startsWith0x && isValidAddress;
+});
+
+export const mustBeEthereumContractAddress = memoize(
+  async (network: string, address: string) => {
+    const provider = getProvider(network) as JsonRpcProvider;
+    const contractCode = await provider.getCode(address);
+
+    return (
+      contractCode && contractCode.replace('0x', '').replace(/0/g, '') !== ''
+    );
+  },
+  (url, contractAddress) => `${url}_${contractAddress}`
+);
+
+export const getContractABI = async (
+  network: string,
+  contractAddress: string
+): Promise<string> => {
+  const apiUrl = EXPLORER_API_URLS[network];
+
+  if (!apiUrl) {
+    return '';
+  }
+
+  const isEthereumAddress = mustBeEthereumAddress(contractAddress);
+  const isEthereumContractAddress = await mustBeEthereumContractAddress(
+    network,
+    contractAddress
+  );
+
+  if (!isEthereumAddress || !isEthereumContractAddress) {
+    return '';
+  }
+
+  try {
+    const { result, status } = await fetchContractABI(apiUrl, contractAddress);
+
+    if (status === '0') {
+      return '';
+    }
+
+    return result;
+  } catch (e) {
+    console.error('Failed to retrieve ABI', e);
+    return '';
+  }
+};
+
+export const isAllowedMethod = ({ name, type }: AbiItem): boolean => {
+  return type === 'function' && !!name;
+};
+
+export const getMethodAction = ({
+  stateMutability
+}: AbiItem): 'read' | 'write' => {
+  if (!stateMutability) return 'write';
+  return ['view', 'pure'].includes(stateMutability) ? 'read' : 'write';
+};
+
+export const extractUsefulMethods = (abi: AbiItem[]): AbiItemExtended[] => {
+  const allowedAbiItems = abi.filter(isAllowedMethod) as AllowedAbiItem[];
```

Comment:
We may want to hide the "view" functions from the list like "totalSupply"
![image](https://user-images.githubusercontent.com/16245250/120926766-829e2480-c708-11eb-9a5e-d7963e22764c.png)

Reply:
I added some logic to only list `write` methods